src/Image.cc

```diff
@@ -799,8 +800,26 @@ void Image::jpegToARGB(jpeg_decompress_struct* args, uint8_t* data, uint8_t* src
 cairo_status_t
 Image::decodeJPEGIntoSurface(jpeg_decompress_struct *args) {
   cairo_status_t status = CAIRO_STATUS_SUCCESS;
-  
-  uint8_t *data = new uint8_t[naturalWidth * naturalHeight * 4];
+
+  size_t stride = naturalWidth * 4;
+  if (naturalWidth != 0 && stride / naturalWidth != 4) {
```

Comment:
We've got a host of these unchecked muls... might be worth copying a version of this function so it's easier to continue addressing them later.

https://github.com/nodejs/node/blob/master/src/util-inl.h#L297-304

(not sure if that link works, github is acting up. Line 297-304.)